run.go

```diff
@@ -0,0 +1,163 @@
+	if err != nil {
+		return fmt.Errorf("Failed to set up Fastlane authentication parameters: %v", err)
+	}
+	var globallySetAuthEnvs []string
+	for envKey, envValue := range authEnvs {
+		if _, set := os.LookupEnv(envKey); set {
+			globallySetAuthEnvs = append(globallySetAuthEnvs, envKey)
+		}
+
+		envs = append(envs, fmt.Sprintf("%s=%s", envKey, envValue))
+	}
+	if len(globallySetAuthEnvs) != 0 {
+		f.logger.Warnf("Fastlane authentication-related environment varibale(s) (%s) are set, overriding.", globallySetAuthEnvs)
+		f.logger.Infof("To stop overriding authentication-related environment variables, please set Bitrise Apple Developer Connection input to 'off' and leave authentication-related inputs empty.")
+	}
+
+	buildlogPth := ""
+	if tempDir, err := pathutil.NormalizedOSTempDirPath("fastlane_logs"); err != nil {
+		f.logger.Errorf("Failed to create temp dir for fastlane logs, error: %s", err)
+	} else {
+		buildlogPth = tempDir
+		envs = append(envs, "FL_BUILDLOG_PATH="+buildlogPth)
+	}
+
+	name := "fastlane"
+	args := opts.LaneOptions
+	options := &command.Opts{
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+		Dir:    opts.WorkDir,
+		Env:    append(os.Environ(), envs...),
+	}
+	var cmd command.Command
+	if opts.UseBundler {
+		cmd = f.rbyFactory.CreateBundleExec(name, args, opts.GemVersions.bundler.Version, options)
+	} else {
+		cmd = f.rbyFactory.Create(name, args, options)
+	}
+
+	f.logger.Donef("$ %s", cmd.PrintableCommandArgs())
+
+	deployDir := os.Getenv("BITRISE_DEPLOY_DIR")
+	if deployDir == "" {
+		f.logger.Warnf("No BITRISE_DEPLOY_DIR found")
+	}
+	deployPth := filepath.Join(deployDir, "fastlane_env.log")
+
+	if err := cmd.Run(); err != nil {
+		f.logger.Println()
+		f.logger.Errorf("Fastlane command: (%s) failed", cmd.PrintableCommandArgs())
+		f.logger.Errorf("If you want to send an issue report to fastlane (https://github.com/fastlane/fastlane/issues/new), you can find the output of fastlane env in the following log file:")
+		f.logger.Println()
+		f.logger.Infof(deployPth)
+		f.logger.Println()
+
+		if fastlaneDebugInfo, err := f.fastlaneDebugInfo(opts.WorkDir, opts.UseBundler, opts.GemVersions.bundler); err != nil {
+			f.logger.Warnf("%s", err)
+		} else if fastlaneDebugInfo != "" {
+			if err := fileutil.WriteStringToFile(deployPth, fastlaneDebugInfo); err != nil {
+				f.logger.Warnf("Failed to write fastlane env log file, error: %s", err)
+			}
+		}
+
+		if err := filepath.Walk(buildlogPth, func(path string, info os.FileInfo, err error) error {
+			if err != nil {
+				return err
+			}
+			if !info.IsDir() {
+				if relLogPath, err := filepath.Rel(buildlogPth, path); err != nil {
+					return err
+				} else if err := os.Rename(path, filepath.Join(deployDir, strings.Replace(relLogPath, "/", "_", -1))); err != nil {
+					return err
+				}
+			}
+			return nil
+		}); err != nil {
+			f.logger.Errorf("Failed to walk directory, error: %s", err)
+		}
+
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
```

Comment:
NOTE: This is an excellent example of why it makes sense to use short variable declaration for errors: the error (`err`) we are handling here is declared at line 87, and between the two, we use the same env var name for errors (`err`), if any didn't use the short variable declaration (but simply reassign the value of err), the original error would be overwritten.

(for this big and risky scope for error might even make sense to use a different variable name (for example: `runErr`/`fastlaneRunErr`...), just to make sure it doesn't get reassigned.)